StreamDeckSDK/ESDFilesystem.h

```diff
@@ -0,0 +1,28 @@
+// @copyright  (c) 2022, Frederick Emmott
+// This source code is licensed under the MIT-style license found in
+// the LICENSE file.
+#pragma once
+
+/* Use std::filesystem where available, ghc::filesystem otherwise. */
+
+#ifdef ESD_HAS_STD_FILESYSTEM
```

Comment:
Thanks - could you make this independent from the CMake with:

1. check if `__has_include(<filesystem>)`
2. if so, `#include <filesystem>`, then `#if __cpp_lib_filesystem`

Ideally, also change the check you have in cmake similarly

Reply:
See above - `ghc_filesystem` specifically recommends the check I used here. I could duplicate the check from the CMakeFile here, but I didn't want to possibly get into the state where CMake and the compiler disagree about which implementation to use, so I used an explicit `target_compile_definitions` so they reuse the same test. (Also, I could push the check *only* into the header, but I assumed you would not want the dependency downloaded or linked if it's not needed.)